File: payroll-api/src/main/java/com/company/payroll/mapper/StaffLeaveDetailsMapper.java
package com.company.payroll.mapper;

import com.company.payroll.model.StaffLeaveDetails;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface StaffLeaveDetailsMapper {
    int deleteByPrimaryKey(Integer ldId);
    int insert(StaffLeaveDetails row);
    int insertSelective(StaffLeaveDetails row);
    StaffLeaveDetails selectByPrimaryKey(Integer ldId);
    List<StaffLeaveDetails> selectList();
    int updateByPrimaryKeySelective(StaffLeaveDetails row);
    int updateByPrimaryKey(StaffLeaveDetails row);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/StaffLeaveDetailsMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.StaffLeaveDetails
import org.springframework.stereotype.Repository

@Repository
interface StaffLeaveDetailsMapper {
  fun deleteByPrimaryKey(ldId: Int): Int

  fun insert(row: StaffLeaveDetails): Int

  fun insertSelective(row: StaffLeaveDetails): Int

  fun selectByPrimaryKey(ldId: Int): StaffLeaveDetails

  fun updateByPrimaryKeySelective(row: StaffLeaveDetails): Int

  fun updateByPrimaryKey(row: StaffLeaveDetails): Int
}